pyradiosky/skymodel.py

```diff
@@ -959,10 +1031,192 @@ def copy(self):
+            raise NotImplementedError(
+                "Healpix map transformations are currently not implemented for catalogs "
+                "with polarization information."
+            )
+        #  quickly check the validity of the transformation using a dummy SkyCoord object.
+        coords = SkyCoord(0, 0, unit="rad", frame=self.frame)
+
+        # we will need the starting frame for some interpolation later
+        old_frame = coords.frame
+
+        coords = coords.transform_to(frame)
+
+        frame = coords.frame
+
+        if not isinstance(frame, (Galactic, ICRS)):
+            raise ValueError(
+                f"Supplied frame {frame.__class__.__name__} is not supported at "
+                "this time. Only 'galactic' and 'icrs' frames are currently supported.",
+            )
+
+        hp_obj_new = astropy_healpix.HEALPix(
+            nside=self.nside,
+            order=self.hpx_order,
+            frame=frame,
+        )
+        hp_obj_old = astropy_healpix.HEALPix(
+            nside=self.nside,
+            order=self.hpx_order,
+            frame=old_frame,
+        )
+
+        # It is not immediately obvious how many unique pixels the output
+        # array will have. Initialize a full healpix map, then we will downselect
+        # later to only valid pixels.
+        out_stokes = units.Quantity(
+            np.zeros((4, self.Nfreqs, hp_obj_new.npix)), unit=self.stokes.unit
+        )
+        # Need the coordinates of the pixel centers in the new frame
+        # then we will use these to interpolate for each freq/stokes
+        new_pixel_locs = hp_obj_new.healpix_to_skycoord(np.arange(hp_obj_new.npix))
+
+        for stokes_ind in range(4):
+            # We haven't implemented a Q+iU rotation fix yet.
+            if stokes_ind > 0:
+                continue
+
+            for freq_ind in range(self.Nfreqs):
+                masked_old_frame = np.ma.zeros(hp_obj_new.npix).astype(
+                    self.stokes.dtype
+                )
+                # Default every pixel to masked, then unmask ones we have data for
+                masked_old_frame.mask = np.ones(masked_old_frame.size).astype(bool)
+                masked_old_frame.mask[self.hpx_inds] = False
+
+                masked_old_frame[self.hpx_inds] = self.stokes[
+                    stokes_ind, freq_ind
+                ].value
+
+                masked_new_frame = hp_obj_old.interpolate_bilinear_skycoord(
+                    new_pixel_locs,
+                    masked_old_frame,
+                )
+
+                out_stokes[stokes_ind, freq_ind] = units.Quantity(
+                    masked_new_frame.data,
+                    unit=self.stokes.unit,
+                )
+
+        # Each frequency/stokes combination should have the same input pixels
+        # and rotations, therefore the output mask should be equivalent.
+        self.hpx_inds = np.nonzero(~masked_new_frame.mask)[0]
```

Comment:
Should we add an option to get back a full sky map rather than a cut sky map? It seems like some people want a full sky map even if it has zeros in it.

Reply:
That's a good idea. I'll add it in.